test-framework/src/test/java/org/apache/brooklyn/test/framework/TestSensorTest.java

```diff
@@ -332,6 +343,56 @@ public void testFailFastIfNoSensor() throws Exception {
         assertStartFails(app, NullPointerException.class, Asserts.DEFAULT_LONG_TIMEOUT);
     }
 
+    @Test
+    public void testDoesNotLogStacktraceRepeatedly() throws Exception {
+        final long time = System.currentTimeMillis();
+        final String sensorValue = String.format("%s%s%s", Identifiers.makeRandomId(8), time, Identifiers.makeRandomId(8));
+        
+        // Test case will repeatedly fail until we have finished our logging assertions.
+        // Then we'll let it complete by setting the sensor.
+        TestSensor testCase = app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TIMEOUT, Asserts.DEFAULT_LONG_TIMEOUT)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, STRING_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, newListAssertion("matches", String.format(".*%s.*", time))));
+
+        String loggerName = Repeater.class.getName();
+        ch.qos.logback.classic.Level logLevel = ch.qos.logback.classic.Level.DEBUG;
+        Predicate<ILoggingEvent> repeatedFailureMsgMatcher = EventPredicates.containsMessage("repeated failure; excluding stacktrace");
+        Predicate<ILoggingEvent> stacktraceMatcher = EventPredicates.containsExceptionStackLine(TestFrameworkAssertions.class, "checkActualAgainstAssertions");
+        Predicate<ILoggingEvent> filter = Predicates.or(repeatedFailureMsgMatcher, stacktraceMatcher);
+        LogWatcher watcher = new LogWatcher(loggerName, logLevel, filter);
+
+        Task<Void> task;
+        watcher.start();
+        try {
```

Comment:
Since `LogWartcher` is  `Closeable` can do this in the `try` block and it will be closed automatically.
```Java
try (LogWatcher watcher = new LogWatcher(loggerName, logLevel, filter)) {
    // stuff
}
```